spec/api-web-frame-spec.js

```diff
@@ -139,4 +139,11 @@ describe('webFrame module', function () {
       webFrame.setLayoutZoomLevelLimits(0, 25)
     })
   })
+
+  it('supports setting cache capacity', function () {
+    assert.doesNotThrow(function () {
+      webFrame.setCacheCapacity(0)
+      webFrame.setCacheCapacity(1024)
```

Comment:
What if it is called with `NaN`, a string, or without arguments at all?
I expect it to be silently ignored. Can you please add more test cases?

Reply:
Do we want it silently ignored instead of explicitly throwing out? for example, I checked existing api `webFrame::setZoomLevel` and putting unexpected value (like `string`type) explicitly throw out like `Uncaught TypeError: Error processing argument at index 0, conversion failure from a` (which currently `setCacheCapacity` does - doesn't accept negative number, or non-number) For me this is kind of first time try to make changes in Electron so not very much aware between api surface behavior consistency, so I may be wrong.

Reply:
At least it shouldn't crash the app.
Throwing exceptions is ok too.

Reply:
![image](https://user-images.githubusercontent.com/1210596/35425688-99494206-0211-11e8-9cb6-97b37d2d8f55.png)

Shaped api more consistently, only allow number greater than 0 and other possible value throws.

Reply:
I believe around the place we've used a boolean return value to indicate success / failure over throwing errors for invalid usage.  I.e. The conversion failure would throw an error, but using an invalid number would return `false`, using a valid number would return `true`.

I don't think this is a strict thing we follow though (it isn't documented anywhere), just would probably prefer the boolean in this case 😄 Will defer to @alexeykuzmin on whether to boolean or throw 👍

Reply:
I would recommend throwing on invalid input and returning a boolean with a status of the whole operation.
If we don't know if the cache capacity was actually changed after the call, there is no reason to return anything.